org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java

```diff
@@ -242,7 +242,7 @@ protected void fillMenu(Menu menu) {
 						populateMenuItem(mode, ext, menu, null, accelerator++);
 					} else if (configurations.length > 0) {
 						for (ILaunchConfiguration configuration : configurations) {
-							populateMenuItem(mode, ext, menu, configuration, -1);
+							populateMenuItem(mode, ext, menu, configuration, accelerator++);
```

Comment:
This was done by intention, there can only be 9 shortcuts and these should be reserved to the "real" shortcuts as one other whise easily runs out of shortcuts.

Reply:
The expanded launch is often as important as the other shortcuts, so it also needs to be easily accessible, as easily as it used to be before we expanded the shortcut to launch.

Reply:
But there is not **one** there can be many, for example m2e adds a lot of default shortcuts and then the available numbers are easily exhausted.

Reply:
Sure there can be many, but still they can be as important as other shortcuts so should be also accessible with mneminics.

Reply:
If you want that you can add a run config to the "favorites" and then you get a shortcut for a run config.

Reply:
>  People who were used to hitting context-menu button, then R then 2 to run a JUnit test will miss it now.

See my comments below, but people that are used to do so might be even confused if hitting `R` then `2` suddenly executes *something else* and **not** a  JUnit test, that's why my initial intend was to not change the assignment but keep it intact with skipping additional (dynamic) run config.